internal/pkg/flink/internal/store/store.go

```diff
@@ -104,39 +104,29 @@ func (s *Store) FetchStatementResults(statement types.ProcessedStatement) (*type
 	}
 
 	// Process remote statements that are now running or completed
-	pageToken := statement.PageToken
-	runningNoTokenRetries := 5
-	for i := 0; i < runningNoTokenRetries; i++ {
-		// TODO: we need to retry a few times on transient errors
-		statementResultObj, err := s.client.GetStatementResults(s.appOptions.GetOrgResourceId(), s.appOptions.GetEnvId(), statement.StatementName, pageToken)
-		if err != nil {
-			return nil, &types.StatementError{Msg: err.Error()}
-		}
+	statementResultObj, err := s.client.GetStatementResults(s.appOptions.GetOrgResourceId(), s.appOptions.GetEnvironmentId(), statement.StatementName, statement.PageToken)
```

Comment:
FYI: If you're using the correct flink-gateway client, it should have built-in retry logic if the backend returns 429 or 5xx.

Reply:
Oh nice!
In this case though, the retry was just there to get around a backend issue (in some cases it didn't return a page token when there should be one. So when that happened, we retried four times to make sure there really is no token)